### PR TITLE
Trace: Fixes Tracing/diagnostics hour-times to 24Hours

### PR DESCRIPTION
**Pull request by kirankumarkolli**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Include samples if adding new API, and include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.TraceTextWriter.cs [hunk 1]
```diff
@@ -177,7 +179,7 @@ private static class HeaderLengths
                 writer.Write("Component");
                 writer.Write(space);
 
-                writer.Write(trace.StartTime.ToString("hh:mm:ss:fff", CultureInfo.InvariantCulture));
+                writer.Write(trace.StartTime.ToString(TraceWriter.HourTimeFormatString, CultureInfo.InvariantCulture));
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.TraceTextWriter.cs:182]: Is Compute depending on this format for anything?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.TraceTextWriter.cs:182]: Unsure but its incorrect right?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.TraceTextWriter.cs:182]: True but my point is, if they currently have any automation component parsing/interpreting this, this change will break it.

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Tracing/TraceWriter.TraceTextWriter.cs:182]: Please check https://github.com/Azure/azure-cosmos-dotnet-v3/pull/3577 

Got into rebase issues and ended-up re-creating a new PR.

**Pull request closed without merging**